src/api/routes/weather.py

```
@@ -0,0 +1,30 @@
+
+router = APIRouter()
+
+@router.get("/weather", response_model=List[WeatherData])
+def get_weather_data(db: Session = next(get_db())):
+    weather_data = db.query(WeatherData).all()
+    if not weather_data:
+        raise HTTPException(status_code=404, detail="No weather data found")
+    return weather_data
+
+@router.post("/weather", response_model=WeatherData)
+def create_weather_data(weather: WeatherData, db: Session = next(get_db())):
```

Ruff (0.12.2)

Do not perform function call `next` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)